host/frontend/host-orchestrator/operator.go

```diff
@@ -46,12 +51,26 @@ func main() {
 
 	setupDeviceEndpoint(pool, config, socketPath)
 	r := setupServerRoutes(pool, polledSet, config)
-
 	http.Handle("/", r)
-	log.Println("Client endpoint created")
-	if err := http.ListenAndServe(fmt.Sprint(":", port), nil); err != nil {
-		log.Fatal("ListenAndServe client: ", err)
-	}
+
+	wg := new(sync.WaitGroup)
+	wg.Add(2)
+
+	go func() {
+		log.Println("Client http endpoint created")
+		log.Fatal(http.ListenAndServe(fmt.Sprint(":", httpPort), nil))
+		wg.Done()
```

Comment:
this should be done with `defer` instead, in case log.Fatal panics for example.

Reply:
Done